### PR TITLE
fix(events): match values in event pattern array are not deduplicated

### PR DESCRIPTION
**Pull request by titanjer**
Since event pattern supports content filtering expression and the expression only allow one key, for example: `{ prefix: '2017-10-02' }`, we can use **JSON.stringify** instead of **deepEqual** to compare element in event pattern array. Please refer to [Content-based Filtering with Event Patterns](https://docs.aws.amazon.com/eventbridge/latest/userguide/content-filtering-with-event-patterns.html).

The following screenshot is the event pattern output when I use `cloudtrail.Trail.onEvent`. First red is the example of duplicate match values. Second red is the example of prefix rule.

Screen Shot 2020-12-02 at 12 26 43 AM

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by shivlaks — DISMISSED** [review]
@titanjer thanks for submitting this fix! feels like a useful one that should be marked as a fix instead of a chore since it represents a bug.

can we also add a test to the `addEventPattern` in `rule` (the usage site)

**Comment by titanjer** [comment]
@shivlaks ok! let me do it

**Comment by titanjer** [comment]
@shivlaks sorry to click `re-request review` button accidentally.

**Comment by titanjer** [comment]
Hi @shivlaks, I modified an `addEventPattern` test in `rule` for ur advice. Moreover, I use another way to de-duplicate the pattern array, not only in concat time but also de-duplicate elements in srcValue and destValue and keep the order.

File: packages/@aws-cdk/aws-events/lib/util.ts [hunk 1]
```diff
@@ -38,9 +38,11 @@ export function mergeEventPattern(dest: any, src: any) {
           `Type mismatch between existing pattern ${JSON.stringify(destValue)} and added pattern ${JSON.stringify(srcValue)}`);
       }
 
-      // if this is an array, concat the values
+      // if this is an array, concat and deduplicate the values
       if (Array.isArray(srcValue)) {
-        destObj[field] = destValue.concat(srcValue);
+        const result = [...destValue, ...srcValue];
+        const resultJson = result.map(i => JSON.stringify(i));
+        destObj[field] = result.filter((value, index) => resultJson.indexOf(JSON.stringify(value)) === index);
```
Comment (titanjer) [inline packages/@aws-cdk/aws-events/lib/util.ts:45]: We use `indexOf` to find the `current value` from start. If the output does not equal to value `index`, it means there is a same value before current value and current value meed to remove.

**Review by shivlaks — APPROVED** [review]

**Pull request merged**